File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/persistence/repository/RecipeEntityRepository.java
package com.tarbadev.witchcraft.recipes.persistence.repository;

import com.tarbadev.witchcraft.recipes.persistence.entity.RecipeEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface RecipeEntityRepository extends JpaRepository<RecipeEntity, Integer> {
  List<RecipeEntity> findAllByOrderByName();

  List<RecipeEntity> findAllByFavorite(Boolean favorite);

  List<RecipeEntity> findTop8ByOrderByIdDesc();
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/recipes/persistence/repository/RecipeEntityRepository.kt
package com.tarbadev.witchcraft.recipes.persistence.repository

import com.tarbadev.witchcraft.recipes.persistence.entity.RecipeEntity
import org.springframework.data.jpa.repository.JpaRepository

interface RecipeEntityRepository : JpaRepository<RecipeEntity, Int> {
    fun findAllByOrderByName(): List<RecipeEntity>

    fun findAllByFavorite(favorite: Boolean?): List<RecipeEntity>

    fun findTop8ByOrderByIdDesc(): List<RecipeEntity>
}
